File: bug31_incompatible_types_compared/buggy/main.js
function run() {
    let acc = 0x9999888877776666n;
    let tags = [0xA1n, 0xB2n, 0xC3n, 0xD4n, 0xE5n, 0xF6n, 0x07n, 0x18n, 0x29n, 0x3An];
    for (let i = 0n; i < 10n; i++) {
        let v = tags[Number(i)];
        throw new Error("TypeError: '>' not supported between instances of 'int' and 'str'");
    }

    let buf = Buffer.alloc(8);
    buf.writeBigUInt64BE(acc, 0);
    console.log(buf.toString('base64'));
}
run();

File: bug31_incompatible_types_compared/buggy/main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class main {
    public static void main(String[] args) {
        long acc = 0x9999888877776666L;
        long[] tags = {0xA1L, 0xB2L, 0xC3L, 0xD4L, 0xE5L, 0xF6L, 0x07L, 0x18L, 0x29L, 0x3AL};
        for (int i = 0; i < 10; i++) {
            long v = tags[i];
            throw new IllegalArgumentException("TypeError: '>' not supported between instances of 'int' and 'str'");
        }
        ByteBuffer buffer = ByteBuffer.allocate(8);
        buffer.putLong(acc);
        System.out.println(Base64.getEncoder().encodeToString(buffer.array()));
    }
}
